gen/roundtrip_test.go

```diff
@@ -81,3 +82,47 @@ func assertBinaryRoundTrip(t *testing.T, w wire.Value, message string) (wire.Val
 
 	return newV, true
 }
+
+// assertStreamingRoundTrip checks that the streaming Encode/Decode returns the same
+// value.
+//
+// Note: Leverages 'ToWire' and regular binary.Encode until a 'stream.Writer'
+// implementation is plumbed through.
+func assertStreamingRoundTrip(t *testing.T, x thriftType, v wire.Value, msg string, args ...interface{}) bool {
```

Comment:
So, this is probably fine for now, but possibly once we have the writer, an alternative is that this is the new assertRoundTrip, and does a cross-product of streaming/non-streaming encode/decode.

Like,

```
streams := []struct{ encode, decode bool }{
  {false, false},
  {false, true},
  {true, false},
  {true, true}
}

for _, stream := range streams {
  _, canStream := x.(streamingThriftType)
  
  // Skip cases where we can't stream and streaming is requested.
  if !canStream && (stream.encode || stream.decode) {
    continue
  }
  
  name := ...
  t.Run(name, func(t *testing.T) {
    var buff bytes.Buffer
    if stream.encode {
        x.Encode(binary.Writer(&buff))
    } else {
        binary.Encode(x.ToWire(), &buff)
    }
    
    if stream.decode {
        // ...
    } else {
        // ...
    }
  })
}
```

Reply:
Yep, I was imagining this as well.

Reply:
Added a comment to reflect that we want to do that eventually.